docs/src/main/asciidoc/kafka-streams-guide.adoc

```diff
@@ -0,0 +1,1141 @@
+
+== Creating the Producer Maven Project
+
+First, we need a new project with the temperature value producer.
+Create a new project with the following command:
+
+[source, subs=attributes+]
+----
+mkdir producer && cd producer && \
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=kafka-streams-quickstart-producer \
+    -Dextensions="kafka" \
+    && cd ..
+----
+
+This command generates a Maven project, importing the Reactive Messaging and Kafka connector extensions.
+
+=== The Temperature Value Producer
+
+Create the `producer/src/main/java/org/acme/quarkus/sample/generator/ValuesGenerator.java` file,
+with the following content:
+
+[source, java]
+----
+package org.acme.quarkus.sample.generator;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.reactivex.Flowable;
+import io.smallrye.reactive.messaging.kafka.KafkaMessage;
+
+/**
+ * A bean producing random temperature data every second.
+ * The values are written to a Kafka topic (temperature-values).
+ * Another topic contains the name of weather stations (weather-stations).
+ * The Kafka configuration is specified in the application configuration.
+ */
+@ApplicationScoped
+public class ValuesGenerator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ValuesGenerator.class);
+
+    private Random random = new Random();
+
+    private List<WeatherStation> stations = Collections.unmodifiableList(
+            Arrays.asList(
+                    new WeatherStation(1, "Hamburg", 13),
+                    new WeatherStation(2, "Snowdonia", 5),
+                    new WeatherStation(3, "Boston", 11),
+                    new WeatherStation(4, "Tokio", 16),
+                    new WeatherStation(5, "Cusco", 12),
+                    new WeatherStation(6, "Svalbard", -7),
+                    new WeatherStation(7, "Porthsmouth", 11),
+                    new WeatherStation(8, "Oslo", 7),
+                    new WeatherStation(9, "Marrakesh", 20)
+            ));
+
+
+    @Outgoing("temperature-values")                             // <1>
+    public Flowable<KafkaMessage<Integer, String>> generate() {
+
+        return Flowable.interval(500, TimeUnit.MILLISECONDS)    // <2>
+                .map(tick -> {
+                    WeatherStation station = stations.get(random.nextInt(stations.size()));
+                    double temperature = new BigDecimal(random.nextGaussian() * 15 + station.averageTemperature)
```

Comment:
The width of the line is a bit too long for the web site.

Reply:
Wrapped it.